services/hh/processor.go

```diff
@@ -23,7 +23,20 @@ const (
 	bytesWrite  = "bytes_write"
 )
 
+// MetaProvider specifies the interface of a module that the Processor may query
+// to determine active node IDs.
+type MetaProvider interface {
+	// Nodes returns the active cluster node IDs
+	Nodes() []uint64
+}
+
+// Processor manages the loading and processing of hinted data. If MetaProvider is non-nil
+// it is queried for active nodes during processing. In this case only if a node ID is active
+// will it be processed. Data for inactive nodes remain on disk, and the purge process must
```

Comment:
Ha, I see that, I didn't notice the comment there. Let me think about moving this functionality into the `purge` call.